plugins/out_kafka/kafka.c

```diff
@@ -92,9 +96,22 @@ int produce_message(struct flb_time *tm, msgpack_object *map,
 
         /* Pack timestamp */
         msgpack_pack_str(&mp_pck, ctx->timestamp_key_len);
```

Comment:
at this point the msgpack already prepared a buffer of ctx->timestamp_key_len bytes, but then in line 107 if ctx->time_key_format is set, it packs again a key header, leading to a corruption in the outgoing message.